app/src/main/java/app/revanced/manager/ui/screen/PatchesSelectorScreen.kt

```diff
@@ -222,102 +261,146 @@ fun PatchesSelectorScreen(
         }
     }
 
-    search?.let { query ->
-        SearchView(
-            query = query,
-            onQueryChange = { search = it },
-            onActiveChange = { if (!it) search = null },
-            placeholder = { Text(stringResource(R.string.search_patches)) }
-        ) {
-            val bundle = bundles[pagerState.currentPage]
-
-            LazyColumnWithScrollbar(
-                modifier = Modifier.fillMaxSize()
-            ) {
-                fun List<PatchInfo>.searched() = filter {
-                    it.name.contains(query, true)
+    Scaffold(
+        topBar = {
+            app.revanced.manager.ui.component.SearchBar(
+                query = query,
+                onQueryChange = setQuery,
+                expanded = searchExpanded,
+                onExpandedChange = setSearchExpanded,
+                placeholder = {
+                    Text(stringResource(R.string.search_patches))
+                },
+                leadingIcon = {
+                    val rotation = remember { Animatable(0f) }
+                    LaunchedEffect(searchExpanded) {
+                        val direction = if (searchExpanded) 360f else 0f
+
+                        rotation.animateTo(
+                            targetValue = direction,
+                            animationSpec = tween(durationMillis = 400, easing = EaseInOut)
+                        )
+                    }
```

Comment:
Wouldn't `animateFloatAsState` work for this?

```kt
val rotation by animateFloatAsState(
	if (searchExpanded) 360f else 0f,
	remember { tween(durationMillis = 400, easing = EaseInOut) }
)
```

Reply:
You're right, this is just a leftover from a different type of animation I was testing with. Why remember the animation spec? First time seeing that, not gonna lie

Reply:
The source code for the `animate*AsState` functions use `remember` or global `private val`s for the default animation spec. Probably doesn't make a difference though.